src/include/platform/ConfigurationManager.h

```diff
@@ -63,17 +63,24 @@ class ConfigurationManager
     };
 
     CHIP_ERROR GetVendorName(char * buf, size_t bufSize);
+    CHIP_ERROR StoreVendorName(const char * buf, size_t bufSize);
     CHIP_ERROR GetVendorId(uint16_t & vendorId);
+    CHIP_ERROR StoreVendorId(uint16_t vendorId);
     CHIP_ERROR GetProductName(char * buf, size_t bufSize);
+    CHIP_ERROR StoreProductName(const char * buf, size_t bufSize);
     CHIP_ERROR GetProductId(uint16_t & productId);
+    CHIP_ERROR StoreProductId(uint16_t productId);
     CHIP_ERROR GetProductRevisionString(char * buf, size_t bufSize);
+    CHIP_ERROR StoreProductRevisionString(const char * buf, size_t bufSize);
     CHIP_ERROR GetProductRevision(uint16_t & productRev);
     CHIP_ERROR GetSerialNumber(char * buf, size_t bufSize, size_t & serialNumLen);
     CHIP_ERROR GetPrimaryWiFiMACAddress(uint8_t * buf);
     CHIP_ERROR GetPrimary802154MACAddress(uint8_t * buf);
     CHIP_ERROR GetManufacturingDate(uint16_t & year, uint8_t & month, uint8_t & dayOfMonth);
     CHIP_ERROR GetFirmwareRevisionString(char * buf, size_t bufSize);
+    CHIP_ERROR StoreFirmwareRevisionString(const char * buf, size_t bufSize);
     CHIP_ERROR GetFirmwareRevision(uint32_t & firmwareRev);
+    CHIP_ERROR StoreFirmwareRevision(uint32_t firmwareRev);
```

Comment:
All of these new arguments are listed as `F` (FIXED) quality and `R` (read-only) in spec:

```
[width="99%",frame="all",options="header", cols="^,<s,^,^,^,^,^,^"]
|===
| ID | Name                    | Type            | Constraint | Quality | Default | Access   | Conformance
| 0  | InteractionModelVersion | uint16          | _all_      | F       | 1       | R V      | M
| 1  | VendorName              | <<string-def>>  | max 32     | F       | MS      | R V      | M
| 2  | VendorID                | <<vendor-id>>   | _all_      | F       | MS      | R V      | M
| 3  | ProductName             | <<string-def>>  | max 32     | F       | MS      | R V      | M
| 4  | ProductID               | uint16          | _all_      | F       | MS      | R V      | M
| 5  | NodeLabel               | <<string-def>>  | max 32     | N       | ""      | RW VM    | M
| 6  | Location                | <<string-def>>  | 2          | N       | ""      | RW VA    | M
| 7  | HardwareVersion         | uint16          | _all_      | F       | 0       | R V      | M
| 8  | HardwareVersionString   | <<string-def>>  | 1 to 64    | F       | MS      | R V      | M
| 9  | SoftwareVersion         | uint32          | _desc_     | F       | 0       | R V      | M
| 10 | SoftwareVersionString   | <<string-def>>  | 1 to 64    | F       | MS      | R V      | M
| 11 | ManufacturingDate       | <<string-def>>  | 8 to 16    | F       | MS      | R V      | O
| 12 | PartNumber              | <<string-def>>  | max 32     | F       | MS      | R V      | O
| 13 | ProductURL              | <<string-def>>  | max 256    | F       | MS      | R V      | O
| 14 | ProductLabel            | <<string-def>>  | max 64     | F       | MS      | R V      | O
| 15 | SerialNumber            | <<string-def>>  | max 32     | F       | MS      | R V      | O
| 16 | LocalConfigDisabled     | bool            | _all_      | N       | False   | RW VM    | O
| 17 | Reachable               | bool            | _all_      |         | True    | R V      | O
| 18 | UniqueID                | <<string-def>>  | max 32     | F       | MS      | R V      | O
|===
```

That means they *cannot* be written, and they have to be fixed for the given version of firmware. These should NOT have writers. In fact, I would argue that we should remove these fields from ConfigurationManager entirely, and define a new interface for BasicInformationCluster to access the necessary settings.